numpyro/contrib/distributions/distribution.py

```diff
@@ -9,10 +9,25 @@
 # Copyright (c) 2003-2019 SciPy Developers.
 # All rights reserved.
 from contextlib import contextmanager
+try:
+    from scipy._lib._util import getargspec_no_self
+except ImportError:
+    from collections import namedtuple
+    from scipy._lib._util import getfullargspec_no_self
```

Comment:
I think if we want this module, we should ideally reimplement these utility functions in contrib, otherwise there is no guarantee that they will continue to exist in future scipy versions.